app/views/users/_form.html.erb

```diff
@@ -69,12 +69,28 @@
     <div class='tab-pane' id='roles'>
       <%= checkbox_f f, :admin if User.current.can_change_admin_flag? %>
       <%= multiple_checkboxes f, :roles, @user, Role.givable.for_current_user,
-                               { :label => _('Roles')}, {:disabled => @editing_self ? Role.givable.for_current_user.pluck(:id) : false } %>
+                              { :label => _('Roles')}, {:disabled => @editing_self ? Role.givable.for_current_user.pluck(:id) : false } %>
+      <div class="form-group">
+        <% if @user.cached_usergroups.uniq.any? %>
```

Comment:
Note that the `cached_usergroups` relation contains all user groups including those that are part of directly associated user groups. That's why we have the cache association there. They might be duplicated since multiple user groups can contain the same user group which is why we need `uniq`.